ee/app/canned-responses/server/hooks/onMessageSentParsePlaceholder.ts

```diff
@@ -28,7 +28,12 @@ const placeholderFields = {
 	},
 };
 
-callbacks.add('beforeSaveMessage', (message: IMessage, room: IOmnichannelRoom): any => {
+callbacks.add('beforeSaveMessage', (message: IMessage): any => {
+	if (!message.msg || message.msg === '') {
+		return message;
+	}
+
+	const room: IOmnichannelRoom = Rooms.findOneById(message.rid);
```

Comment:
If a room object is expected but not provided, we should return IMO.
```suggestion
	if (!room || !isOmnichannelRoom(room)) {
```

Reply:
Actually I went with this approach because I thought it would be the safest considering I found some inconsistencies in the codebase. When I checked all the places throughout the codebase which is running `beforeSaveMessage` hook, all of them expect one passes the second room param. For reference please see the image below.
![image](https://user-images.githubusercontent.com/34130764/128305536-4e4a42e5-f6ce-4d93-9827-84a63bce64f5.png)

Hence due to this inconsistency, I though it would be best if we used the first param to resolve the room object as it would work for all the hooks. 

Let me know if you still think its better to go with the approach you suggested. Thanks :smile:

Reply:
Implemented the following logic
> try to use the room param when available and let's try to find the room if the param is not there or is invalid